lightning/src/chain/channelmonitor.rs

```diff
@@ -2216,6 +2240,27 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 					self.lockdown_from_offchain = true;
 					if *should_broadcast {
 						self.broadcast_latest_holder_commitment_txn(broadcaster, logger);
+						// If the channel supports anchor outputs, we'll need to emit an external
+						// event to be consumed such that a child transaction is broadcast with a
+						// high enough feerate for the parent commitment transaction to confirm.
+						if self.onchain_tx_handler.opt_anchors() {
```

Comment:
Note, iirc `should_broadcast` doesn't mean there are effectively pending HTLCs making the confirmation of this transaction an emergency. In fact, even if we force-close, we could still be satisfied by a slow confirmation and save on fees ? Maybe `ChannelForceClosed` should be enriched with a `bool pending_htlcs` argument

Reply:
In fact, ideally we would return back the amount of the HTLCs at stake, and the user would commit the fee-bumping values they can afford in function of their appreciation of the risk loss. IIRC, LND does something similar, or at least consider the values of the HTLC before to decide on an unilateral closing. I think this is a type of advanced APIs that could be interesting to offer to our users in the future.

In the meanwhile I think it could be good to track follow-ups in an issue.

Reply:
Ah I think we have a confusion here between the `ChannelForceClosed` flag `should_broadcast` and the function `should_broadcast_holder_commitment_txn` called in `block_confirmed`? For the latter, I agree if there is no pending HTLC we''ll never end up broadcasting through this flow. For the former I think the slow confirmation semantic can be captured by the new `pending_htlcs`. I think comment can be updated to indicate that if `pending_htlcs.is_empty()`, confirmation is not time-sensitive.

Reply:
Yes, I think this is okay that `OnchainTxHandler` returns a reasonable default fee bumping strategy, based on `ConfirmationTarget::HighPriority`. Indeed, the event consumer could choose to ignore the feerate and employ their own fee bumping strategy based on historical network mempool data and loss aversion of pending HTLCs. 

> when confirmation is not even an urgency (in the case of an empty commitment).

Sounds fee bumping could be turned off in case of empty commitment. Or at least an option, if the user would like to get back commitment balance liquidity quickly.